sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestRecord.java

```diff
@@ -226,9 +228,19 @@ public long transportRequestId() {
     // region Methods
 
     public boolean expire() {
-        final GoneException error = new GoneException(this.toString(), null, this.args.physicalAddress());
-        BridgeInternal.setRequestHeaders(error, this.args.serviceRequest().getHeaders());
+        final CosmosException error;
+        if (this.args.serviceRequest().isReadOnly()) {
+            // Always converting RequestTimeoutException to GoneException for ReadOnly request. Since they can be safely retried.
+            error = new GoneException(this.toString(), null, this.args.physicalAddress());
+        } else if (!this.hasSendingRequestStarted()) {
+            // If request not sent for write request, converting to GoneException so it can be retried.
+            error = new GoneException(this.toString(), null, this.args.physicalAddress());
```

Comment:
GoneException in the cosmos context means that "The requested resource is no longer available at the server". Why are we reusing that exception here? I understand that GoneException is wired properly for retries but would'nt it make sense to have a client specific exception here which will make it easier to debug?
Thoughts?

Reply:
@mbhaskar  I agree with u, we should have a better definition of the GoneException. Currently, we are kind a little bit abuse the use of it, basically we are using it to wire up the retries, address refresh etc. I think this should be a larger scope discussion about the design etc.

As for debugging, this one can be quickly identified through the  Rntbd diagnostics. But would a different substatus code help here?

Reply:
Yeah different substatus is definitely a good start to reduce the abuse. Eventually we can also introduce a new exception subclassing GoneException for the client generated scenarios

Reply:
Discussed offline, agreed will leave this change out of this PR. 

Meanwhile, @mbhaskar created this working item to track it : https://github.com/Azure/azure-sdk-for-java/issues/20271

Thanks